app/filtering.js

```diff
@@ -520,14 +521,29 @@ function registerForDownloadListener (session) {
     }
 
     const defaultPath = path.join(getSetting(settings.DOWNLOAD_DEFAULT_PATH) || getSetting(settings.DEFAULT_DOWNLOAD_SAVE_PATH) || app.getPath('downloads'), itemFilename)
-    const savePath = ((process.env.SPECTRON || (!getSetting(settings.DOWNLOAD_ALWAYS_ASK) && !item.promptForSaveLocation())) ? defaultPath : dialog.showSaveDialog(win, { defaultPath }))
+    let savePath = ((process.env.SPECTRON || (!getSetting(settings.DOWNLOAD_ALWAYS_ASK) && !item.promptForSaveLocation())) ? defaultPath : dialog.showSaveDialog(win, { defaultPath }))
 
     // User cancelled out of save dialog prompt
     if (!savePath) {
       event.preventDefault()
       return
     }
 
+    if (!getSetting(settings.DOWNLOAD_ALWAYS_ASK) && !item.promptForSaveLocation()) {
+      let willNotOverwrite = false
```

Comment:
double negative checks are also hard to understand `!willNotOverwrite` -> `willOverwrite`